### PR TITLE
Add aws-lambda-powertools

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

_Created with Grayskull_

### CODE REVIEW

File: recipes/aws-lambda-powertools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    - poetry >=0.12
+    - python >=3.6,<3.9
+  run:
+    - aws-xray-sdk >=2.5.0,<3
+    - boto3 >=1.12,<2
+    - python-fastjsonschema >=2.14.5,<3
+    - jmespath >=0.10.0,<0.11
+    - python >=3.6,<3.9
+  run_constrained:
+    - email-validator
```
Comment (BastianZim) [inline recipes/aws-lambda-powertools/meta.yaml:30]: `email-validator` was listed as an optional dependency (https://github.com/awslabs/aws-lambda-powertools-python/blob/2f840bfbaa3390ddaa81aa2b34f17832e4ef5e3c/pyproject.toml#L28) but with no specific version. 
I'd therefore exclude it again, correct?

**Comment by BastianZim** [comment]
@conda-forge/help-python Would be ready for review, thanks!

File: recipes/aws-lambda-powertools/meta.yaml (hunk 1 above)
Comment (chrisburr) [reply recipes/aws-lambda-powertools/meta.yaml:30]: Yup, it won't do anything if it's not got a version constraint associated

Comment (BastianZim) [reply recipes/aws-lambda-powertools/meta.yaml:30]: Ok will do, thanks for the confirmation! Any idea why the added it in the first place?

**Comment by BastianZim** [comment]
Thank you!

**Pull request merged**